### PR TITLE
chore(*): bump golang version to 1.13.6

### PR DESCRIPTION
**Pull request by iawia002**
**What this PR does / why we need it**:

* bump golang version to 1.13.6
* go mod related settings
* bump golangci-lint version to 1.23.6

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @bbbmj 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -112,8 +115,9 @@ build-linux:
 	  -e GOOS=linux                                                                    \
 	  -e GOARCH=amd64                                                                  \
 	  -e GOPATH=/go                                                                    \
+	  -e GOFLAGS=$(GOFLAGS)                                                            \
```
Comment (supereagle) [inline Makefile:119]: Is this flag not needed for `build-local`?

Comment (iawia002) [reply Makefile:119]: Yes, but I have exported the `GOFLAGS` environment variable above for the local environment

**Comment by supereagle** [comment]
LGTM

**Comment by supereagle** [comment]
ping @bbbmj

File: Makefile [hunk 2]
```diff
@@ -72,6 +72,9 @@ CPUS ?= $(shell /bin/bash hack/read_cpus_available.sh)
 # Track code version with Docker Label.
 DOCKER_LABELS ?= git-describe="$(shell date -u +v%Y%m%d)-$(shell git describe --tags --always --dirty)"
 
+# This will force go to use the vendor files instead of using the `$GOPATH/pkg/mod`. (vendor mode)
+export GOFLAGS := -mod=vendor
```
Comment (bbbmj) [inline Makefile:77]: Can you provide more information about this flag?

File: Makefile [hunk 3]
```diff
@@ -91,7 +95,7 @@ lint: $(GOLANGCI_LINT)
 	@$(GOLANGCI_LINT) run
 
 $(GOLANGCI_LINT):
-	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(BIN_DIR) v1.20.1
+	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(BIN_DIR) v1.23.6
```
Comment (zhujian7) [inline Makefile:98]: Why do we need this change?

Comment (iawia002) [reply Makefile:98]: General update for performance and bug fixes

**Comment by supereagle** [comment]
@bbbmj  PTAL

File: Makefile (hunk 2 above)
Comment (hicaistar) [reply Makefile:77]: any conclusion? Must we add it?

Comment (iawia002) [reply Makefile:77]: I have added more info about this flag in 6cc695153a217455e5ba6b8ae6055fc63d2fdd3e, and yes you must add it

Comment (hicaistar) [reply Makefile:77]: > I have added more info about this flag in [6cc6951](https://github.com/caicloud/golang-template-project/commit/6cc695153a217455e5ba6b8ae6055fc63d2fdd3e), and yes you must add it

👌

**Pull request closed without merging**